From 1f056007af3943e2d15fc4dfecb57aa2aef7008e Mon Sep 17 00:00:00 2001
From: William Wu <william.wu@rock-chips.com>
Date: Sun, 28 Jun 2020 11:49:53 +0800
Subject: [PATCH] Revert "phy: add cp_test callback"

This reverts commit 955bb70bf167976ee9e57ccd54a613ae3a038948.

Fix the following changed report by build_abi.sh for GKI

'struct phy_ops at phy.h:56:1' changed:
  type size changed from 512 to 576 (in bits)
  1 data member insertion:
    'int (phy*)* phy_ops::cp_test', at offset 448 (in bits) at phy.h:70:1
  there are data member changes:
    'module* phy_ops::owner' offset changed from 448 to 512 (in bits) (by +64 bits)
  17 impacted interfaces

Change-Id: I5f163b804aaff2fe9c6e3a7c054de3f8df0c791c
Signed-off-by: William Wu <william.wu@rock-chips.com>
---
 drivers/phy/phy-core.c  | 15 ---------------
 include/linux/phy/phy.h | 10 ----------
 2 files changed, 25 deletions(-)

diff --git a/drivers/phy/phy-core.c b/drivers/phy/phy-core.c
index 4e3e4657c087..8c7e6b443c39 100644
--- a/drivers/phy/phy-core.c
+++ b/drivers/phy/phy-core.c
@@ -392,21 +392,6 @@ int phy_reset(struct phy *phy)
 }
 EXPORT_SYMBOL_GPL(phy_reset);
 
-int phy_cp_test(struct phy *phy)
-{
-	int ret;
-
-	if (!phy || !phy->ops->cp_test)
-		return 0;
-
-	mutex_lock(&phy->mutex);
-	ret = phy->ops->cp_test(phy);
-	mutex_unlock(&phy->mutex);
-
-	return ret;
-}
-EXPORT_SYMBOL_GPL(phy_cp_test);
-
 int phy_calibrate(struct phy *phy)
 {
 	int ret;
diff --git a/include/linux/phy/phy.h b/include/linux/phy/phy.h
index e549ecc1b74b..5f7ed6d86e43 100644
--- a/include/linux/phy/phy.h
+++ b/include/linux/phy/phy.h
@@ -56,7 +56,6 @@ enum phy_mode {
  * @set_mode: set the mode of the phy
  * @reset: resetting the phy
  * @calibrate: calibrate the phy
- * @cp_test: prepare for the phy compliance test
  * @owner: the module owner containing the ops
  */
 struct phy_ops {
@@ -67,7 +66,6 @@ struct phy_ops {
 	int	(*set_mode)(struct phy *phy, enum phy_mode mode);
 	int	(*reset)(struct phy *phy);
 	int	(*calibrate)(struct phy *phy);
-	int	(*cp_test)(struct phy *phy);
 	struct module *owner;
 };
 
@@ -176,7 +174,6 @@ static inline enum phy_mode phy_get_mode(struct phy *phy)
 }
 int phy_reset(struct phy *phy);
 int phy_calibrate(struct phy *phy);
-int phy_cp_test(struct phy *phy);
 static inline int phy_get_bus_width(struct phy *phy)
 {
 	return phy->attrs.bus_width;
@@ -303,13 +300,6 @@ static inline int phy_reset(struct phy *phy)
 	return -ENOSYS;
 }
 
-static inline int phy_cp_test(struct phy *phy)
-{
-	if (!phy)
-		return 0;
-	return -ENOSYS;
-}
-
 static inline int phy_calibrate(struct phy *phy)
 {
 	if (!phy)
-- 
2.35.3

